### ISSUE TITLE
Awesome paper, testing OM on my detector right now, still have some questions!

### ISSUE THREAD

**Issue by OE-jimvanoosten** [ZhuHaoranEIS/Orthogonal-FGOD#2]
Very interesting paper. Deeply impressed with the idea of OM. Training my two-stage detector with OM at the moment. Very curious whether it will help!

For my understanding:

_We replace the last convolutional layer or fully connected
layer in the original head with OM to introduce orthogonal
constraints. Specifically, we predefine the orthogonal feature
prototypes v_k for each class k ∈ {1, ..., C} (C means the
number of categories). For a network’s output feature u, The
probability that the feature u belongs to class j is given by
calculating the cosine similarity between u and v_j . In this way,
features of the same category (j) exhibit similar expressions
(all lie within the space of v_j), while features of different
categories (i, j) occupy mutually orthogonal spaces (vi ⊥ vj )._

In layman's terms: each class is assigned an orthogonal feature (which somehow I keep interpreting as a fixed rotation angle), and then the cosine similarity (please explain) between the extracted feature and the predefined orthogonal feature is calculated. I guess the FC layer basically learns to recognize an object by comparing the predefined feature with the extracted one.

Love to hear from you guys! Thanksssss

**Comment by ZhuHaoranEIS** [comment]
> Very interesting paper. Deeply impressed with the idea of OM. Training my two-stage detector with OM at the moment. Very curious whether it will help!
> 
> For my understanding:
> 
> _We replace the last convolutional layer or fully connected layer in the original head with OM to introduce orthogonal constraints. Specifically, we predefine the orthogonal feature prototypes v_k for each class k ∈ {1, ..., C} (C means the number of categories). For a network’s output feature u, The probability that the feature u belongs to class j is given by calculating the cosine similarity between u and v_j . In this way, features of the same category (j) exhibit similar expressions (all lie within the space of v_j), while features of different categories (i, j) occupy mutually orthogonal spaces (vi ⊥ vj )._
> 
> In layman's terms: each class is assigned an orthogonal feature (which somehow I keep interpreting as a fixed rotation angle), and then the cosine similarity (please explain) between the extracted feature and the predefined orthogonal feature is calculated. I guess the FC layer basically learns to recognize an object by comparing the predefined feature with the extracted one.
> 
> Love to hear from you guys! Thanksssss

We are glad that our work could be of help to you. As you mentioned, most of the explanations are accurate. In our response, we have further elaborated on why the method is effective, and we hope this provides you with some useful guidance.
[Uploading ORTH_GRSL_Response.pdf…]()

**Comment by ZhuHaoranEIS** [comment]
[ORTH_GRSL_Response.pdf](https://github.com/user-attachments/files/16993936/ORTH_GRSL_Response.pdf)

**Comment by ZhuHaoranEIS** [comment]
> Very interesting paper. Deeply impressed with the idea of OM. Training my two-stage detector with OM at the moment. Very curious whether it will help!
> 
> For my understanding:
> 
> _We replace the last convolutional layer or fully connected layer in the original head with OM to introduce orthogonal constraints. Specifically, we predefine the orthogonal feature prototypes v_k for each class k ∈ {1, ..., C} (C means the number of categories). For a network’s output feature u, The probability that the feature u belongs to class j is given by calculating the cosine similarity between u and v_j . In this way, features of the same category (j) exhibit similar expressions (all lie within the space of v_j), while features of different categories (i, j) occupy mutually orthogonal spaces (vi ⊥ vj )._
> 
> In layman's terms: each class is assigned an orthogonal feature (which somehow I keep interpreting as a fixed rotation angle), and then the cosine similarity (please explain) between the extracted feature and the predefined orthogonal feature is calculated. I guess the FC layer basically learns to recognize an object by comparing the predefined feature with the extracted one.
> 
> Love to hear from you guys! Thanksssss

We are glad that our work could be of help to you. As you mentioned, most of the explanations are accurate. In our response, we have further elaborated on why the method is effective, and we hope this provides you with some useful guidance.
[ORTH_GRSL_Response.pdf](https://github.com/user-attachments/files/16993939/ORTH_GRSL_Response.pdf)
